src/lib.rs

```diff
@@ -126,7 +142,30 @@ impl Uart {
         }
     }
 
-    /// TODO: implement init
+    /// Initializes PL011 UART.
+    /// clk: Uart clock in Hz.
+    pub fn init(&mut self, clock: u32, baud_rate: u32) {
+        let divisor = (clock << 2) / baud_rate;
+
+        // SAFETY: self.registers points to the control registers of a PL011 device which is
+        // appropriately mapped, as promised by the caller of `Uart::new`.
+        unsafe {
+            // Disable UART before programming.
+            let mut cr: Control = addr_of_mut!((*self.registers).cr).read_volatile();
+            cr &= !Control::UARTEN;
+            addr_of_mut!((*self.registers).cr).write_volatile(cr);
+            // Program IBRD.
```

Comment:
Nit: it would be more helpful for these comments to explain what IBRD and FBRD stand for.